src/server/handler/request.rs

```diff
@@ -239,6 +289,221 @@ impl Server {
             error: None,
         });
     }
+
+    pub(crate) fn handle_references(&mut self, id: RequestId, params: ReferenceParams) {
```

Comment:
What are your thoughts on having `handle_references` return an `Option::<Vec<Location>>`? 

This should allow the try operator to do early returns on a `None`, then `self.respond_empty` can be handled in a wrapper function.